packages/core/src/storages/sitemap_request_list.ts

```diff
@@ -0,0 +1,292 @@
+import { parseSitemap } from '@crawlee/utils';
+import ow from 'ow';
+
+import { KeyValueStore } from './key_value_store';
+import type { IRequestList } from './request_list';
+import { purgeDefaultStorages } from './utils';
+import { Request } from '../request';
+
+export interface SitemapRequestListOptions {
+    sitemapUrls: string[];
+    proxyUrl?: string;
+    persistStateKey?: string;
+}
+
+interface SitemapRequestListState {
+    reclaimed: string[];
+    handledSitemapUrls: string[];
+    handledUrls: string[];
+    currentSitemapUrl?: string;
+    currentSitemapUrlQueue?: string[];
+}
+
+/**
+ * A list of URLs to crawl parsed from a sitemap.
+ *
+ * The loading of the sitemap is performed in the background so that crawling can start before the sitemap is fully loaded.
+ */
+export class SitemapRequestList implements IRequestList {
+    /**
+     * Set of URLs that were returned by fetchNextRequest() and not marked as handled yet.
+     * @internal */
+    inProgress = new Set<string>();
+
+    /** Set of URLs for which reclaimRequest() was called. */
+    private reclaimed = new Set<string>();
+
+    /** Request objects for in-progress and reclaimed URLs so that we can keep track of state */
+    private requestData = new Map<string, Request>();
+
+    /**
+     * URLs of
+     * 1. sitemaps that have been completely processed and
+     * 2. processed sitemap items of the current (partially processed) sitemap
+     */
+    private handledUrls = { sitemapUrls: new Set<string>(), urls: new Set<string>() };
+
+    /** URLs loaded from the sitemaps */
+    private queuedUrlsBySitemap = new Map<string, string[]>();
+
+    /** Number of URLs that were marked as handled */
+    private handledUrlCount = 0;
+
+    /** Indicates whether the background processing of sitemap contents has already finished.  */
+    private isSitemapFullyLoaded = false;
+
+    private persistStateKey?: string;
+
+    private store?: KeyValueStore;
+
+    private sitemapUrls: string[];
+
+    private proxyUrl: string | undefined;
+
+    /** @internal */
+    private constructor(options: SitemapRequestListOptions) {
+        ow(
+            options,
+            ow.object.exactShape({
+                sitemapUrls: ow.array,
+                proxyUrl: ow.optional.string,
+                persistStateKey: ow.optional.string,
+            }),
+        );
+
+        this.persistStateKey = options.persistStateKey;
+        this.sitemapUrls = options.sitemapUrls;
+        this.proxyUrl = options.proxyUrl;
+    }
+
+    private startLoadingInBackground(): void {
```

Comment:
i would prefer having this method async and ignoring the promise at the call site instead of this

Reply:
Does ignoring mean "not awaiting" or something more intricate?

Reply:
yes, not awaiting, and to tell eslint we actually do that on purpose we use `void` keyword:

```ts
void this.startLoadingInBackground();
```

(but then the name is not really matching what it does :])